pkgs/applications/misc/waybar/default.nix

```diff
@@ -0,0 +1,54 @@
+{ 
+stdenv, fetchFromGitHub, makeWrapper, meson, pkgconfig, ninja, 
+gtkmm3, jsoncpp, libinput, libsigcxx, wayland, wlroots, fmt
+, traySupport  ? true,  libdbusmenu-gtk3 ? null
+, pulseSupport ? false, libpulseaudio    ? null
+, nlSupport    ? true,  libnl            ? null
+, swaySupport  ? false, sway             ? null
+}:
+
+assert traySupport -> libdbusmenu-gtk3 != null;
+assert pulseSupport  -> libpulseaudio != null;
+assert nlSupport -> libnl != null;
+assert swaySupport -> sway != null;
```

Comment:
There’s no need for this. You don’t need to provide default values for the package parameters.